### DISCUSSION TITLE
Launching apps via AGS makes them depend on AGS process

### DISCUSSION THREAD

**Discussion by zabihexes** [Aylur/ags#242] [Ideas]
Application service not disowning the apps when launching seems like slight issue maybe.

This way if AGS needs to be restarted or it dies for whatever reason then all apps launched by it get killed too.

What do you think?

**Comment by kotontrion** [comment]
This is a known issue, with no solution yet. There is a workaround though. instead of `app.launch()`you could use this:
```js
Hyprland.sendMessage(`dispatch exec ${app.executable}`);
app.frequency++;
```

**Reply by zabihexes** [reply]
I see. Thanks

**Reply by zabihexes** [reply]
~~Seems to break some apps. Like they think I am trying to open some non existing file(`%F` for example) if I launch them this way.~~

~~Dunno what that is.~~

Ended up with this which seems to work perfect so far.

```ts
await Hyprland.sendMessage(`dispatch exec gtk-launch "${app.desktop}"`);
```

**Comment by Aylur** [comment]
Thats just how Gtk works, if you want it to be detached, avoid `app.launch` and use some other means like what kotontrion mentioned
